js/packages/components/shared-components/AnimatedNumber.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { FC, useEffect, useRef, useState } from 'react'
+import { Text, View } from 'react-native'
+import Animated, { EasingNode } from 'react-native-reanimated'
+
+const NUMBERS: number[] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+const usePrevious = (value: number) => {
+	const ref = useRef<number>()
+	useEffect(() => {
+		ref.current = value
+	})
+
+	if (typeof ref.current === 'undefined') {
+		return 0
+	}
+
+	return ref.current
+}
+
+export const AnimatedNumber: FC<{
+	number: number
+	fontStyle?: any
+	animationDuration?: number
+	easing?: Animated.EasingNodeFunction
+}> = ({ number, fontStyle, animationDuration, easing }) => {
+	const prevNumber = usePrevious(number)
+	const prevNumberString = String(Math.abs(prevNumber))
+
+	const animateToNumberString = String(Math.abs(number))
+	const animateToNumbersArr = Array.from(animateToNumberString, Number)
+	const prevNumberersArr = Array.from(prevNumberString, Number)
+
+	const [numberHeight, setNumberHeight] = useState<number>(0)
+	const animations = animateToNumbersArr.map((__, index) => {
+		if (typeof prevNumberersArr[index] !== 'number') {
+			return new Animated.Value(0)
+		}
+
+		const animationHeight = -1 * (numberHeight * prevNumberersArr[index])
+		return new Animated.Value(animationHeight)
+	})
+
+	useEffect(() => {
+		animations.map((animation, index) => {
+			if (typeof animateToNumbersArr[index] !== 'number') {
+				return
+			}
+
+			Animated.timing(animation, {
+				toValue: -1 * (numberHeight * animateToNumbersArr[index]),
+				duration: animationDuration || 1400,
+				easing: easing || EasingNode.elastic(1.2),
+			}).start()
+		})
+	}, [number, animateToNumbersArr, animationDuration, animations, easing, numberHeight])
+
+	const getTranslateY = (index: number) => {
```

Comment:
Sorry i forgot this function, same for the memoized hook, i think we have to use a `useCallback` here to

Reply:
Thank you for the constructive review <3